### PR TITLE
sclang: Simplify TreeView signals

### PR DESCRIPTION
**Pull request by scztt**
TreeView redirected it's signals in a non-straightfoward way so it could receive and send QcTreeWidget::ItemPtr (which represent individual items in the tree). But, signals can cast types, so all we need here is (a) a cast operator for our ItemPtr, and (b) forwarding for all the signals on the base class. Rather than changing names of singals as before, I just kept them consistant with the base class and updated the TreeView.sc to match. We get a few more for free that no one bothered to expose previously.

### CODE REVIEW

File: SCClassLibrary/Common/GUI/Base/enums.sc [hunk 1]
```diff
@@ -253,3 +253,13 @@ QToolButtonStyle {
 
 	*new { arg name; ^this.perform(name) }
 }
+
+QScrollHint {
+	classvar
```
Comment (telephon) [inline SCClassLibrary/Common/GUI/Base/enums.sc]: these could be `const`.

Comment (scztt) [reply SCClassLibrary/Common/GUI/Base/enums.sc]: All the enum values in enums.sc should be const - but, in general, the enums stuff needs a little clean-up and factoring. Mind if I address separately? I'd rather match style and implementation with this checkin, and deal with enum issues all at once.

**Comment by telephon** [comment]
yes, sure!

**Comment by telephon** [comment]
(I mean, not at all, please go ahead)

File: SCClassLibrary/Common/GUI/Base/QTreeView.sc [hunk 2]
```diff
@@ -1,6 +1,5 @@
 TreeView : View {
-	var <itemPressedAction;
-	var <onItemChanged;
+	var <onItemExpanded, <onItemCollapsed, <onItemPressed, <onItemClicked, <onCurrentItemChanged;
```
[the remark this answers is not in the document]
Comment (brianlheim) [reply SCClassLibrary/Common/GUI/Base/QTreeView.sc]: what change in particular? new methods are added but it seems the behavior of `itemPressedAction` is still the same, it retrieves the appropriate callback

File: QtCollider/widgets/QcTreeWidget.h [hunk 3]
```diff
@@ -90,13 +104,22 @@ class QcTreeWidget : public QTreeWidget
...
-  void action();
-  void itemPressedAction();
-  void currentItemChanged();
-  void expanded(QcTreeWidget::ItemPtr);
-  void collapsed(QcTreeWidget::ItemPtr);
+Q_SIGNALS:
+  void itemPressed(QcTreeWidget::ItemPtr item, int column);
+  void itemClicked(QcTreeWidget::ItemPtr item, int column);
+  void itemDoubleClicked(QcTreeWidget::ItemPtr item, int column);
+  void action(QcTreeWidget::ItemPtr item, int column); //    void itemActivated(QTreeWidgetItem *item, int column)
```
Comment (brianlheim) [inline QtCollider/widgets/QcTreeWidget.h]: what's the point of this comment?

File: SCClassLibrary/Common/GUI/Base/enums.sc [hunk 4]
```diff
@@ -253,3 +253,13 @@ QToolButtonStyle {
...
+	<ensureVisible = 0,
+	<positionAtTop = 1,
+	<positionAtBottom = 2,
+	<positionAtCenter = 3;
+
+	*new { arg name; ^this.perform(name) }
+}
```
Comment (brianlheim) [inline SCClassLibrary/Common/GUI/Base/enums.sc]: missing newline

File: QtCollider/widgets/QcTreeWidget.h (hunk 3 above)
Comment (scztt) [reply QtCollider/widgets/QcTreeWidget.h]: That signal is redirected from a different signal from the base class. Added clarification to comment.

**Comment by muellmusik** [comment]
What is the status of this PR? Just awaiting conflict resolution?

**Comment by brianlheim** [comment]
> What is the status of this PR? Just awaiting conflict resolution?

yes. this was pre-reformat, so using the `rebase` subcommand of clang-format.py would fix it.

https://github.com/supercollider/supercollider/wiki/Cpp-formatting-instructions#i-made-a-pr-before-the-reformat-and-it-has-merge-conflicts-now-how-do-i-fix-it

**Comment by brianlheim** [comment]
closing as stale, please rebase and reopen if you have time @scztt

**Pull request closed without merging**